spec/consensus/light-client.md

```diff
@@ -60,259 +61,368 @@ For the purpose of this lite client specification, we assume that the Tendermint
+Towards Lite Client Completeness:
+- The check is successful if sufficiently many validators of *h1* are still validators in *h2* and signed *h2*.
+- If *h2.Header.height = h1.Header.height + 1*, and both headers were generated correctly, the test passes
 
-_Correctness arguments_
+*Verification Condition:* We may need a Tendermint invariant stating that if *h2.Header.height = h1.Header.height + 1* then *signers(h2.Commit) \subseteq h1.Header.NextV*.
 
-Towards Lite Client Accuracy:
+*Remark*: The variable *trustThreshold* can be used if the user believes that relying on one correct validator is not sufficient. However, in case of (frequent) changes in the validator set, the higher the *trustThreshold* is chosen, the more unlikely it becomes that CheckSupport returns true for non-adjacent headers.
 
-- Assume by contradiction that _h2_ was not generated correctly and the lite client sets trust to true because _CheckSupport_ returns true.
-- h1 is trusted and sufficiently new
-- by Tendermint Fault Model, less than 1/3 of voting power held by faulty validators => at least one correct validator _v_ has signed _h2_.
-- as _v_ is correct up to now, it followed the Tendermint consensus protocol at least up to signing _h2_ => _h2_ was correctly generated, we arrive at the required contradiction.
+**VerifyHeader.** The function *VerifyHeader* captures high level logic, i.e., application call to the lite client module to (optionally download) and
+verify header for some height. The core verification logic is captured by *CanTrust* function that iteratively try to establish trust in given header
+by relying on *CheckSupport* function.
 
-Towards Lite Client Completeness:
 
-- The check is successful if sufficiently many validators of _h1_ are still validators in _h2_ and signed _h2_.
-- If _h2.Header.height = h1.Header.height + 1_, and both headers were generated correctly, the test passes
+```go
+func VerifyHeader(height, trustThreshold) error {
+  if h2, exists := Store.Get(height); exists {
+    if isWithinTrustedPeriod(h2) { return nil }
+    return ErrHeaderNotWithinTrustedPeriod(h2)
+  }
 
-_Verification Condition:_ We may need a Tendermint invariant stating that if _h2.Header.height = h1.Header.height + 1_ then _signers(h2.Commit) \subseteq h1.Header.NextV_.
+  h2 := Commit(height)
+  if !verify(h2) { return ErrInvalidHeader(h2) }
+  if !isWithinTrustedPeriod(h2) { return ErrHeaderNotWithinTrustedPeriod(h2) }
 
-_Remark_: The variable _trustlevel_ can be used if the user believes that relying on one correct validator is not sufficient. However, in case of (frequent) changes in the validator set, the higher the _trustlevel_ is chosen, the more unlikely it becomes that CheckSupport returns true for non-adjacent headers.
+  // get the highest trusted headers lower than h2
+  h1 = Store.HighestTrustedSmallerThan(height)
```

Comment:
Suggestion:
 - rename `Store` to `TrustedStore`
 - rename `HighestTrustedSmallerThan` to `HighestSmallerThan` or simply `GetSmallerOrEqual` (we already use `Get` somewhere else and this would be slightly more consistent).

Reply:
> Also, what happens if such a header was not stored yet

Oh, I see. It will return `nil` then (in the next line). Would also be good if these cases were treated consistently.